src/lib/eina/eina_thread_win32.c

```diff
@@ -0,0 +1,188 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2020 Carlos Ré Signor
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+#include "eina_config.h"
+#include "eina_lock.h"
+#include "eina_sched.h"
+#include "eina_cpu.h"
+#include "eina_thread.h"
+/* undefs EINA_ARG_NONULL() so NULL checks are not compiled out! */
+#include "eina_safety_checks.h"
+
+#include "eina_debug_private.h"
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#include <Windows.h>
+#include <errno.h>
+#include <string.h>
+
+typedef struct _Eina_win32_thread_func
+{
+   void *data;
+   void *(*func)(void *data);
+
+} Eina_win32_thread_func;
+
+typedef struct _Eina_win32_thread_attr
+{
+   LPSECURITY_ATTRIBUTES lpThreadAttributes;
+   SIZE_T dwStackSize;
+   DWORD dwCreationFlags;
+} Eina_win32_thread_attr;
```

Comment:
I think because of EFL conventions you should use:
```suggestion
typedef struct _Eina_Win32_Thread_Func
{
   void *data;
   void *(*func)(void *data);

} Eina_Win32_Thread_Func;

typedef struct _Eina_Win32_Thread_Attr
{
   LPSECURITY_ATTRIBUTES lpThreadAttributes;
   SIZE_T dwStackSize;
   DWORD dwCreationFlags;
} Eina_Win32_Thread_Attr;
```